src/wallet/CarbonWallet.ts

```diff
@@ -324,6 +329,15 @@ export class CarbonWallet {
     return wallet;
   }
 
+  public static withRainbowKit(rainbowKit: RainbowKitAccount, evmChainId: string, compressedPubKeyBase64: string, addressOptions: SWTHAddressOptions, walletProvider: string, opts: Omit<CarbonWalletInitOpts, "signer"> = {}) {
+    const signer = RainbowKitAccount.createRainbowKitSigner(rainbowKit, evmChainId, compressedPubKeyBase64, addressOptions)
+    const wallet = CarbonWallet.withSigner(signer, compressedPubKeyBase64, {
+      providerAgent: walletProvider,
+      ...opts,
```

Comment:
Since this function is especially for RainbowKit wallet, think you can declare `isRainbowKit: true` directly in the function instead of asking user to do that.

i.e.
```typescript
const wallet = CarbonWallet.withSigner(signer, compressedPubKeyBase64, {
  ...opts,
  providerAgent: walletProvider,
  isRainbowKit: true,
});
```